### PR TITLE
cloud dm: add troubleshooting tips

### PR DESCRIPTION
**Pull request by hfxsd**
### What is changed, added or deleted? (Required)

Added troubleshooting tips for Data Migration on TiDB Cloud.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 1]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+If you encounter any problems during the migration, you can refer to the following solutions.
+
+- Error message: "The required binary log for migration no longer exists on the source database. Please make sure binary log files are kept for long enough time for migration to succeed."
+
+    This error means that the binlogs to be migrated has been cleaned up and can only be restored by creating a new task.
+
+    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs.
+
+- Error message: "Failed to dial the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
```
Comment (D3Hunter) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:212]:
```suggestion
- Error message: "Failed to connect the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 2]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+## Troubleshooting
+
+If you encounter any problems during the migration, you can refer to the following solutions.
+
+- Error message: "The required binary log for migration no longer exists on the source database. Please make sure binary log files are kept for long enough time for migration to succeed."
+
+    This error means that the binlogs to be migrated has been cleaned up and can only be restored by creating a new task.
+
+    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs.
```
Comment (D3Hunter) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:210]:
```suggestion
    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs if it's needed by some migration job.
```

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Comment by D3Hunter** [comment]
@frank945946 ptal

**Review by Frank945946 — APPROVED** [review]
LGTM

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 3]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "The required binary log for migration no longer exists on the source database. Please make sure binary log files are kept for long enough time for migration to succeed."
+
+    This error means that the binlogs to be migrated has been cleaned up and can only be restored by creating a new task.
+
+    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs if it's needed by some migration job.
+
+- Error message: "Failed to connect to the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
+
+    This error means that the connection to the upstream database failed. Check whether the upstream database is started and can be connected to using the specified parameters. After confirming that the upstream database is available, you can try to recover the task by using `restart`.
```
Comment (ran-huang) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:214]: 这里是指点击界面上的 Restart 按钮，还是指运行 restart 命令？
```suggestion
    This error means that the connection to the upstream database failed. Check whether the upstream database is started and can be connected to using the specified parameters. After confirming that the upstream database is available, you can try to recover the task by clicking **Restart**.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 4]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+    Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by running `restart`.
+
+- Error message: "TiDB cluster storage is not enough. Please increase the node storage of TiKV."
+
+    The TiDB cluster storage is running low. It is recommended to [increase the TiKV node storage](/tidb-cloud/scale-tidb-cluster.md#increase-node-storage) and then resume the task by running `restart`.
+
+- Error message: "Failed to connect to the source database. Please check whether the database is available or the maximum connections have been reached."
+
+    Failed to connect to the upstream database, it is recommended to check whether the upstream database is started, the number of database connections has not reached the upper limit, and you can connect using the parameters specified by the job. After confirming that the source database is available, you can try to resume the job by running `restart`.
```
Comment (ran-huang) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:230]:
```suggestion
    Failed to connect to the upstream database. It is recommended to check whether the upstream database is started, the number of database connections has not reached the upper limit, and you can connect using the parameters specified by the job. After confirming that the source database is available, you can try to resume the job by running `restart`.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 5]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
 If a migration job has failed, you can restart it after solving the problem.
 
 You can delete a migration job in any status.
+
+## Troubleshooting
```
Comment (ran-huang) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:202]: source database 和 upstream database 在这篇文档里是同一个意思吗？

Comment (hfxsd) [reply tidb-cloud/migrate-from-mysql-using-data-migration.md:202]: 嗯是的。我统一下

File: tidb-cloud/migrate-from-mysql-using-data-migration.md (hunk 3 above)
Comment (D3Hunter) [reply tidb-cloud/migrate-from-mysql-using-data-migration.md:214]: 界面上的 `restart`按钮，目前云上用户只能从 ui 上操作

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 6]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs if it's needed by some migration job.
+
+- Error message: "Failed to connect to the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
+
+    This error means that the connection to the upstream database failed. Check whether the upstream database is started and can be connected to using the specified parameters. After confirming that the upstream database is available, you can try to recover the task by using `restart`.
+
+- The migration task is interrupted and contains the error "driver: bad connection" or "invalid connection"
+
+    This error means that the connection to the downstream TiDB cluster failed. Check whether the downstream TiDB cluster is in `normal` state and can be connected with the username and password specified by the job. After confirming that the downstream TiDB cluster is available, you can try to resume the task by running `restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:218]:
```suggestion
    This error means that the connection to the downstream TiDB cluster failed. Check whether the downstream TiDB cluster is in `normal` state and can be connected with the username and password specified by the job. After confirming that the downstream TiDB cluster is available, you can try to resume the task by clicking `Restart`.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 7]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "Failed to connect to the TiDB cluster using the given user and password. Please make sure TiDB Cluster is up and can be connected to using the given user and password."
+
+    Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by running `restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:222]:
```suggestion
    Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by clicking `Restart`.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 8]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "TiDB cluster storage is not enough. Please increase the node storage of TiKV."
+
+    The TiDB cluster storage is running low. It is recommended to [increase the TiKV node storage](/tidb-cloud/scale-tidb-cluster.md#increase-node-storage) and then resume the task by running `restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:226]:
```suggestion
    The TiDB cluster storage is running low. It is recommended to [increase the TiKV node storage](/tidb-cloud/scale-tidb-cluster.md#increase-node-storage) and then resume the task by clicking `Restart`.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 9]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+    Ensure that the binlogs required for incremental migration exist. It is recommended to configure `expire_logs_days` to extend the duration of binlogs. Do not use `purge binary log` to clean up binlogs if it's needed by some migration job.
+
+- Error message: "Failed to connect to the source database using given parameters. Please make sure the source database is up and can be connected using the given parameters."
+
+    This error means that the connection to the source database failed. Check whether the source database is started and can be connected to using the specified parameters. After confirming that the source database is available, you can try to recover the task by clicking **Restart**.
+
+- The migration task is interrupted and contains the error "driver: bad connection" or "invalid connection"
+
+    This error means that the connection to the downstream TiDB cluster failed. Check whether the downstream TiDB cluster is in `normal` state and can be connected with the username and password specified by the job. After confirming that the downstream TiDB cluster is available, you can try to resume the task by clicking `Restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:218]:
```suggestion
    This error means that the connection to the downstream TiDB cluster failed. Check whether the downstream TiDB cluster is in `normal` state and can be connected with the username and password specified by the job. After confirming that the downstream TiDB cluster is available, you can try to resume the task by clicking **Restart**.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 10]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "Failed to connect to the TiDB cluster using the given user and password. Please make sure TiDB Cluster is up and can be connected to using the given user and password."
+
+    Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by clicking `Restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:222]:
```suggestion
    Failed to connect to TiDB cluster. It is recommended to check whether the TiDB cluster is in `normal` state and you can connect with the username and password specified by the job. After confirming that the TiDB cluster is available, you can try to resume the task by clicking **Restart**.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 11]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "TiDB cluster storage is not enough. Please increase the node storage of TiKV."
+
+    The TiDB cluster storage is running low. It is recommended to [increase the TiKV node storage](/tidb-cloud/scale-tidb-cluster.md#increase-node-storage) and then resume the task by clicking `Restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:226]:
```suggestion
    The TiDB cluster storage is running low. It is recommended to [increase the TiKV node storage](/tidb-cloud/scale-tidb-cluster.md#increase-node-storage) and then resume the task by clicking **Restart**.
```

File: tidb-cloud/migrate-from-mysql-using-data-migration.md [hunk 12]
```diff
@@ -198,3 +198,33 @@ You can pause or delete a migration job when it is running.
...
+
+- Error message: "Failed to connect to the source database. Please check whether the database is available or the maximum connections have been reached."
+
+    Failed to connect to the source database. It is recommended to check whether the source database is started, the number of database connections has not reached the upper limit, and you can connect using the parameters specified by the job. After confirming that the source database is available, you can try to resume the job by clicking `Restart`.
```
Comment (hfxsd) [inline tidb-cloud/migrate-from-mysql-using-data-migration.md:230]:
```suggestion
    Failed to connect to the source database. It is recommended to check whether the source database is started, the number of database connections has not reached the upper limit, and you can connect using the parameters specified by the job. After confirming that the source database is available, you can try to resume the job by clicking **Restart**.
```

**Review by ran-huang — APPROVED** [review]

**Pull request merged**